### DISCUSSION TITLE
Deploying page from GH Actions with locked branch

### DISCUSSION THREAD

**Discussion by leweex95** [mkdocs/mkdocs#3736] [Q&A]
I currently have a master and a gh-pages branch in my repository. I am deploying new versions of my branch using Github Actions in an automated manner, at every time a merge of a PR happens. 

Now I want to open up my repository to other contributors and, naturally, I want to make it impossible for them to modify either the master or the gh-pages branches directly. I tried locking the gh-pages branch, for instance, but even the `mkdocs gh-deploy --force` command fails as it cannot write the gh-pages branch if it is locked.

What workarounds and best practices can you recommend in such a scenario?

**Comment by pawamoy** [comment]
By opening your repository to other contributors you mean adding them as collaborators? Surely if you do that, it means that you trust them enough (and guide them enough) not to randomly push to the gh-branch, right?

If you mean to accept PRs, without adding collaborators, then nobody can push to your branches anyway (they have to go through the PR process), so you don't have to lock them.

Otherwise, maybe your GitHub action could use the GitHub API to unlock the branch, push to it, and relock it afterwards. Or maybe there settings to allow force pushes from the repository owners.
